src/book/02-projects/01-nubots/03-build-system.mdx

```diff
@@ -5,3 +5,265 @@ title: Build System
+| --------- | ------------------------------------------------- |
+| configure | Configure the CMake system for the current target |
+| build     | Build the currently configured code               |
+| edit      | Edit configuration files                          |
+| format    | Applies code formatting rules to codebase         |
+| install   | Install build code to specified robot             |
+| module    | Generate new modules                              |
+| run       | Run a role locally                                |
+| target    | Select a target to work on                        |
+
+##### Configure
+
+`./b configure` will ensure that a build folder is created in the current Docker image and will then run `cmake` on the source code ensuring that CMake creates all of the necessary build files inside of the build folder.
+
+It is possible to specify extra arguments to `cmake` to control the configuration process. For example, to enable the building of unit tests one would run
+
+```sh
+./b configure -- -DBUILD_TESTS=ON
+```
+
+---
+
+**NOTE**
+`--` is needed to prevent python from trying to process the argument itself
+
+---
+
+Passing `-i` to `./b configure` will open up an interactive window with `ccmake` so that you may modify the CMake configuration.
+
+##### Build
+
+As one might expect, `./b build` will build all enabled modules using `ninja`.
+
+If unit tests are enabled in the CMake configuration (`-DBUILD_TESTS=ON`) then `./b build test` can be run (after first running `/b build`) to execute all of the unit tests.
+
+##### Edit
+
+If you are running roles on your local machine and you need to edit the configuration or data files for the modules in the role then the preferred method is to edit the file in the source tree and then rebuild the code using `./b build` to copy the edited file into the build folder.
+
+However, some modules generate their configuration file at build-time, making it impossible to edit the file in the source tree. In this situation, you can use `./b edit` to open an interactive window with a text editor allowing you to edit the configuration file in the build tree.
+
+One module that generates it configuration file at build time is `support::loggingg::DataLogging`, to edit the configuration file for `DataLogging` run
+
+```sh
+./b edit config/DataLogging.yaml
+```
+
+The `edit` command uses the editor that is defined in your host shell (using the `EDITOR` environment variable). If `EDITOR` is not set then it will default to `nano`. Currently, the only supported editors are `vim` and `nano`, with `nano` used as the fallback in all cases.
+
+##### Format
+
+NUbots uses three different code formatters. `clang-format` is used to format C++ and protobuf files, `black` and `isort` are used to format python files, and `cmake-format` is used to format CMake files.
+
+If you want to ensure that all files in the codebase are formatted according to our defined style run
```

Comment:
Maybe add that if you use our visual studio it will mostly apply formatting by itself